presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -278,6 +279,133 @@ public void createPartitionedTable(HiveStorageFormat storageFormat)
+                ", _partition_string VARCHAR" +
+                ", _partition_varchar VARCHAR(65535)" +
+                ", _partition_tinyint TINYINT" +
+                ", _partition_smallint SMALLINT" +
+                ", _partition_integer INTEGER" +
+                ", _partition_bigint BIGINT" +
+                ", _partition_decimal_short DECIMAL(3,2)" +
+                ", _partition_decimal_long DECIMAL(30,10)" +
+                ") " +
+                "WITH (" +
+                "partitioned_by = ARRAY[ '_partition_string', '_partition_varchar', '_partition_tinyint', '_partition_smallint', '_partition_integer', '_partition_bigint', '_partition_decimal_short', '_partition_decimal_long' ]" +
+                ") ";
+        assertUpdate(createTable);
+
+        TableMetadata tableMetadata = getTableMetadata(catalog, TPCH_SCHEMA, "test_partitioned_table_pattern");
+
+        List<String> partitionedBy = ImmutableList.of("_partition_string", "_partition_varchar", "_partition_tinyint", "_partition_smallint", "_partition_integer", "_partition_bigint", "_partition_decimal_short", "_partition_decimal_long");
+        assertEquals(tableMetadata.getMetadata().getProperties().get(PARTITIONED_BY_PROPERTY), partitionedBy);
+        for (ColumnMetadata columnMetadata : tableMetadata.getColumns()) {
+            boolean partitionKey = partitionedBy.contains(columnMetadata.getName());
+            assertEquals(columnMetadata.getComment(), annotateColumnComment(Optional.empty(), partitionKey));
+        }
+
+        assertColumnType(tableMetadata, "_string", createUnboundedVarcharType());
+        assertColumnType(tableMetadata, "_partition_string", createUnboundedVarcharType());
+        assertColumnType(tableMetadata, "_partition_varchar", createVarcharType(65535));
+
+        @Language("SQL") String createTableSingleLike = "" +
+                "CREATE TABLE test_partitioned_table_single_like (" +
+                "LIKE test_partitioned_table_pattern " + likeSuffix +
+                ") ";
+        assertUpdate(createTableSingleLike);
+
+        tableMetadata = getTableMetadata(catalog, TPCH_SCHEMA, "test_partitioned_table_single_like");
+
+        if (checkPartition) {
+            assertEquals(tableMetadata.getMetadata().getProperties().get(PARTITIONED_BY_PROPERTY), partitionedBy);
+            for (ColumnMetadata columnMetadata : tableMetadata.getColumns()) {
+                boolean partitionKey = partitionedBy.contains(columnMetadata.getName());
+                assertEquals(columnMetadata.getComment(), annotateColumnComment(Optional.empty(), partitionKey));
+            }
+        }
+
+        assertColumnType(tableMetadata, "_string", createUnboundedVarcharType());
+        assertColumnType(tableMetadata, "_partition_string", createUnboundedVarcharType());
+        assertColumnType(tableMetadata, "_partition_varchar", createVarcharType(65535));
+
+        @Language("SQL") String createTableMixLike = "" +
```

Comment:
These names "single like" and "mix like" are not good. Both versions only have a single "like".